doc/source/user_guide/ug_case_settings.rst

```diff
@@ -370,6 +370,7 @@ domain_nml
    "``maskhalo_bound``", "logical", "mask unused halo cells for boundary updates", "``.false.``"
    "``max_blocks``", "integer", "maximum number of blocks per MPI task for memory allocation", "-1"
    "``nprocs``", "integer", "number of processors to use", "-1"
+   "", "``-1``", "find number of processors automatically", ""
```

Comment:
I think while we're in here, "MPI ranks" would be more accurate than "processors" (for both the new line and the existing one).

Reply:
We seem to use "processors" throughout the rest of the docs, although they seem light on for details about MPI in general. 

I definitely out of my depth here .. but google implies the rank is just the unique name for each process ? So this should be number of "processes" as it could be possible to have multiple processes per processor ! (And by this logic, we should change all the other "processors" to "processes" in the docs)

Reply:
I think "MPI tasks" is technically correct here.  Processors and processes are both wrong.  "MPI ranks" is probably the same thing, but not terminology we generally use in the document.  Lets go with "MPI tasks"?

Reply:
Is "Number of MPI tasks (typically 1 per processor) to use" good?

There are lots of uses of "processor" in the docs, I guess these could be changed to task? Should I make a new issue?

Reply:
"typically 1 per processor" would not be a good thing to add.  The way I think about this is that processors are the hardware in use while tasks and threads are how those processors are used.  So nprocs is specifically associated with the number of MPI tasks only.  It has nothing to do with threads.  If you are running 16 mpi tasks threaded 2 ways each, you are using 32 processors and have 16 MPI tasks.  nprocs is MPI tasks and has nothing to do with the "processors" or "processes" used/set.  To me, processes is a word separate from hardware.  If you could oversubscribe threads on a processors, you might have 1 processor but 4 processes on that processor.  

Again, this is how I think about things and probably how most of the documentation uses that terminology.  I fully acknowledge that my ideas about this may not be the norm.  In that case, we should update the documentation using more standard language.  But before we do, we should carefully think about the terminology.  At this point, I would just update this one line of documentation unless you see other glaring errors.  We could open an issue, but before we do, I'd like for someone to review the documentation (at least a few sections here and there) to see whether there seems to be a bunch of incorrect terminology.  I prefer to have an issue that says "several things need to be fixed such as...." rather than "review documentation for correct/consistent use of tasks, threads, processors, processes".

Reply:
Thanks Tony. Ill just make it say "Number of MPI tasks". I am a bit out of my depth re the best terminology for the documentation, so ill leave it for now.

Reply:
I think you still need to update the documentation here.  Will start some testing on the changes.

Reply:
I made the update, I wonder if it no longer makes sense if compiling with the 'serial' folder instead of the 'mpi' folder ?